### PR TITLE
Add documentation for JSON parsing

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
As per https://github.com/solana-labs/solana/pull/27191#issuecomment-1217627061, there is nothing outside of code that documents for which accounts and instructions RPC nodes will return JSON parsing.

#### Summary of Changes
Add a section to the JSON-RPC docs with additional info and support table.

### CODE REVIEW

File: docs/src/developing/clients/jsonrpc-api.md [hunk 1]
```diff
@@ -205,6 +205,28 @@ Many methods that take a commitment parameter return an RpcResponse JSON object
...
 - `value` : The value returned by the operation itself.
 
+#### Parsed Responses
+
+Some methods support an `encoding` parameter, and can return account or
+instruction data in parsed JSON format if `"encoding":"jsonParsed"` is requested
+and the node has a parser for the owning program. Solana nodes currently support
+JSON parsing for the following native and SPL programs:
+
+| Program | Account State | Instructions |
```
Comment (jstarry) [inline docs/src/developing/clients/jsonrpc-api.md:215]: How about a column for notes on the release versions that support for various states/instructions was added in?

Comment (CriesofCarrots) [reply docs/src/developing/clients/jsonrpc-api.md:215]: Not a bad idea. Can we merge this as-is and then add the version column? I'd rather not dig through github history for all these at the moment.

**Review by jstarry — DISMISSED** [review]

File: docs/src/developing/clients/jsonrpc-api.md (hunk 1 above)
Comment (jstarry) [reply docs/src/developing/clients/jsonrpc-api.md:215]: Yup, no problem

Comment (jstarry) [reply docs/src/developing/clients/jsonrpc-api.md:215]: Version is only useful it's a recent version in beta still so anything stable and older is fine to label as "Stable"

Comment (CriesofCarrots) [reply docs/src/developing/clients/jsonrpc-api.md:215]: Okay, fair enough. Updated.

**Pull request merged**